e2e/interchaintestv8/multichain_test.go

```diff
@@ -1285,3 +1285,313 @@ func (s *MultichainTestSuite) TestTransferEthToCosmosToCosmosAndBack_Groth16() {
+
+	var simdASendTxHash []byte
+	s.Require().True(s.Run("Send from SimdA to SimdB", func() {
+		timeout := uint64(time.Now().Add(30 * time.Minute).Unix())
+
+		transferPayload := transfertypes.FungibleTokenPacketData{
+			Denom:    transferCoin.Denom,
+			Amount:   transferCoin.Amount.String(),
+			Sender:   simdAUser.FormattedAddress(),
+			Receiver: simdBUser.FormattedAddress(),
+			Memo:     "",
+		}
+
+		encodedPayload, err := transfertypes.EncodeABIFungibleTokenPacketData(&transferPayload)
+		s.Require().NoError(err)
+
+		payload := channeltypesv2.Payload{
+			SourcePort:      transfertypes.PortID,
+			DestinationPort: transfertypes.PortID,
+			Version:         transfertypes.V1,
+			Encoding:        transfertypes.EncodingABI,
+			Value:           encodedPayload,
+		}
+
+		resp, err := s.BroadcastMessages(ctx, simdA, simdAUser, 2_000_000, &channeltypesv2.MsgSendPacket{
+			SourceClient:     ibctesting.SecondClientID,
+			TimeoutTimestamp: timeout,
+			Payloads:         []channeltypesv2.Payload{payload},
+			Signer:           simdAUser.FormattedAddress(),
+		})
+		s.Require().NoError(err)
+		s.Require().NotEmpty(resp.TxHash)
+
+		simdASendTxHash, err = hex.DecodeString(resp.TxHash)
+		s.Require().NoError(err)
+	}))
+
+	denomOnSimdB := transfertypes.NewDenom(
+		transferCoin.Denom,
+		transfertypes.NewHop(transfertypes.PortID, ibctesting.SecondClientID),
+	)
+	s.Require().True(s.Run("Receive packet on SimdB", func() {
+		var txBodyBz []byte
+		s.Require().True(s.Run("Retrieve relay tx to SimdB", func() {
+			resp, err := s.RelayerClient.RelayByTx(context.Background(), &relayertypes.RelayByTxRequest{
+				SrcChain:       simdA.Config().ChainID,
+				DstChain:       simdB.Config().ChainID,
+				SourceTxIds:    [][]byte{simdASendTxHash},
+				TargetClientId: ibctesting.SecondClientID,
+			})
+			s.Require().NoError(err)
+			s.Require().NotEmpty(resp.Tx)
+			s.Require().Empty(resp.Address)
+
+			txBodyBz = resp.Tx
+		}))
+
+		s.Require().True(s.Run("Broadcast relay tx on SimdB", func() {
+			_ = s.BroadcastSdkTxBody(ctx, simdB, s.SimdBRelayerSubmitter, 2_000_000, txBodyBz)
+		}))
+
+		s.Require().True(s.Run("Verify balances on SimdB", func() {
+			resp, err := e2esuite.GRPCQuery[banktypes.QueryBalanceResponse](ctx, simdB, &banktypes.QueryBalanceRequest{
+				Address: simdBUser.FormattedAddress(),
+				Denom:   denomOnSimdB.IBCDenom(),
+			})
+			s.Require().NoError(err)
+			s.Require().NotNil(resp.Balance)
+			s.Require().Equal(transferAmount, resp.Balance.Amount.BigInt())
+			s.Require().Equal(denomOnSimdB.IBCDenom(), resp.Balance.Denom)
+		}))
+	}))
+
+	var simdBTransferTxHash []byte
+	s.Require().True(s.Run("Transfer tokens from SimdB to Eth", func() {
+		timeout := uint64(time.Now().Add(30 * time.Minute).Unix())
+		transferCoin := sdk.NewCoin(denomOnSimdB.IBCDenom(), sdkmath.NewIntFromBigInt(transferAmount))
+		transferPayload := transfertypes.FungibleTokenPacketData{
+			Denom:    denomOnSimdB.Path(), // XXX: IBCDenom()?
```

Comment:
This is correct.
Although, we could have sent the transfer using MsgTransfer, in which case we would have used IBCDenom(). I don't think we need to change to do that now. But would be good to at least use MsgTransfer in some other test to verify that it works as expected.
```suggestion
			Denom:    denomOnSimdB.Path(),
```